### PR TITLE
Remove the use of `KafkaProducer` from the dispatcher and use `ReactiveKafkaProducer` instead

### LINKED ISSUE #3191: The base dispatcher module still use `vertx.kafka.client` lib
**Describe the bug**
In the progress of #2928 the base dispatcher dependency on vertx kafka client lib has to remove.
The PR #3177 migrate the base dispatcher to use the custom ReactiveKafkaConsumer. But the KafkaProducer is still there.

**Exit Criteria**
The base dispatcher should be out of vertx kafka dependency and use custom Reactive interfaces instead.

### PR DESCRIPTION
**Pull request by debasishbsws**
Fixes #3191 and progress in #2928

## Proposed Changes

A clean, without-conflict clone PR of #3192

### CODE REVIEW

File: data-plane/THIRD-PARTY.txt [hunk 1]
```diff
@@ -36,12 +36,13 @@ Lists of 229 third-party dependencies.
...
+     (Unknown license) receiver-vertx (dev.knative.eventing.kafka.broker:receiver-vertx:1.0-SNAPSHOT - no url defined)
+     (The Apache Software License, Version 2.0) CloudEvents - API (io.cloudevents:cloudevents-api:2.3.0 - https://cloudevents.github.io/sdk-java/cloudevents-api/)
+     (The Apache Software License, Version 2.0) CloudEvents - Core (io.cloudevents:cloudevents-core:2.3.0 - https://cloudevents.github.io/sdk-java/cloudevents-core/)
+     (The Apache Software License, Version 2.0) CloudEvents - Vert.x Http Binding (io.cloudevents:cloudevents-http-vertx:2.3.0 - https://cloudevents.github.io/sdk-java/cloudevents-http-vertx/)
+     (The Apache Software License, Version 2.0) CloudEvents - JSON Jackson (io.cloudevents:cloudevents-json-jackson:2.3.0 - https://cloudevents.github.io/sdk-java/cloudevents-json-jackson/)
+     (The Apache Software License, Version 2.0) CloudEvents - Kafka Binding (io.cloudevents:cloudevents-kafka:2.3.0 - https://cloudevents.github.io/sdk-java/cloudevents-kafka/)
+     (The Apache Software License, Version 2.0) cloudevents-sql (io.cloudevents:cloudevents-sql:2.3.0 - https://cloudevents.github.io/sdk-java/cloudevents-sql/)
```
Comment (pierDipi) [inline data-plane/THIRD-PARTY.txt:45]: Why this CE SDK version change?

File: data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/MockReactiveKafkaProducer.java
Comment (pierDipi) [inline data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/MockReactiveKafkaProducer.java:1]: Can't we reuse the existing mock and factory?

File: data-plane/THIRD-PARTY.txt (hunk 1 above)
Comment (debasishbsws) [reply data-plane/THIRD-PARTY.txt:45]: There was a merger conflict and the CE SDK versions were changed in #3183. 
from My end, I have just added
```
(Unknown license) receiver-vertx (dev.knative.eventing.kafka.broker:receiver-vertx:1.0-SNAPSHOT - no url defined)
```

While resolving the merge conflict all of these are showing, I don't know why.

File: data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/MockReactiveKafkaProducer.java
Comment (debasishbsws) [reply data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/MockReactiveKafkaProducer.java:1]: Okay I have changed to use the existing receiver Mock into dispatcher module

File: data-plane/THIRD-PARTY.txt (hunk 1 above)
Comment (pierDipi) [reply data-plane/THIRD-PARTY.txt:45]: You might need to rebase the latest main branch

Comment (debasishbsws) [reply data-plane/THIRD-PARTY.txt]: Oh, wait. I mistakenly removed the upgrades version. while resolving conflict.

Comment (debasishbsws) [reply data-plane/THIRD-PARTY.txt]: Done.

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**